src/providers/wfs/qgswfsdataitems.cpp

```diff
@@ -120,7 +121,10 @@ void QgsWfsLayerItem::copyStyle()
   mdata->setData( QGSCLIPBOARD_STYLE_MIME, style.body.toByteArray() );
   mdata->setText( style.body.toString() );
   // Copies data in text form as well, so the XML can be pasted into a text editor
+#ifdef Q_OS_LINUX
   clipboard->setMimeData( mdata, QClipboard::Clipboard );
+#endif
+  clipboard->setMimeData( mdata, QClipboard::Selection );
```

Comment:
What's the reason for the special handling on linux here?

When using linux, does that first set the mimeData to Clipboard and then directly to Selection?

Wondering if we can just go with one of the two which works everywhere or if this might need an `#else` instead.

Reply:
@m-kuhn I just read a comment in the qgsclipboard.cpp which says 
```
// Copy text into the clipboard

  // With qgis running under Linux, but with a Windows based X
  // server (Xwin32), ::Selection was necessary to get the data into
  // the Windows clipboard (which seems contrary to the Qt
  // docs). With a Linux X server, ::Clipboard was required.
  // The simple solution was to put the text into both clipboards.
```
seems like it doesn't work as expected in mac when we only set the mimedata to clipboard

Reply:
Ah, I was assuming that it's a setter for one mimeData property of the clipboard but apparently this can set multiple ones and the second parameter is a hint. This will teach me not to read the docs.

I think the proper way would be to use

```
clipboard->setMimeData( mdata, QClipboard::Clipboard );
if ( clipboard->supportsSelection() )
  clipboard->setMimeData( mdata, QClipboard::Selection );
```

The `#ifdef` Linux is probably not the correct thing to use (there's X for other platforms than linux and there's other things than X on linux, like wayland).